runtime/outer-PEC.js

```diff
@@ -125,6 +143,22 @@ class OuterPEC extends PEC {
     };
   }
 
+  static async _applyResolveRecipe(arc, recipe) {
```

Comment:
I broke this out just for readability as the method above was getting unwieldy. Open to alternate code style opinions of course.